ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewManager.java

```diff
@@ -260,18 +260,52 @@ public void receiveCommand(ReactViewGroup root, int commandId, @Nullable Readabl
     }
   }
 
+  private int getMarkedViewCount(ReactViewGroup parent) {
+    int cnt = 0;
+    Integer tmpInteger = new Integer(1);
+    for (int i = 0; i < parent.mDeleteMark.size(); i++) {
+      if (parent.mDeleteMark.get(i).equals(tmpInteger)) {
+        ++cnt;
+      }
+    }
+    return cnt;
+  }
+
+  private int getPrefixMarkedViewCount(ReactViewGroup parent, int index) {
+    int cnt = 0;
+    Integer tmpInteger = new Integer(1);
+    for (int i = 0; i < parent.mDeleteMark.size() && i <= index + cnt; i++) {
+      if (parent.mDeleteMark.get(i).equals(tmpInteger)) {
+        ++cnt;
+      }
+    }
+    return cnt;
+  }
+
   @Override
   public void addView(ReactViewGroup parent, View child, int index) {
     boolean removeClippedSubviews = parent.getRemoveClippedSubviews();
     if (removeClippedSubviews) {
       parent.addViewWithSubviewClippingEnabled(child, index);
     } else {
-      parent.addView(child, index);
+      parent.mDeleteMark.add(index + getPrefixMarkedViewCount(parent, index),
```

Comment:
the usage of getPrefixMarkedViewCount() here makes addView() inefficient, now we iterate for every "marked child", and we do this twice. can we avoid this?